sympy/utilities/misc.py

```diff
@@ -355,7 +355,8 @@ def translate(s, a, b=None, c=None):
 
     mr = {}
     if a is None:
-        assert c is None
+        if c is not None:
+            raise TypeError('Called translate with 4 arguments and a second argument of None, expected 3 arguments if second argument is None')
```

Comment:
I wasn't sure about your policy for breaking error messages.  I know some projects don't break them so they can be grepped for.  Is SymPy's policy to just break the string across lines?